src/metabase/search/config.clj

```diff
@@ -81,7 +81,10 @@
 
 (assert (= all-models (set models-search-order)) "The models search order has to include all models")
 
-(def ^:private default-weights
+(def ^{:private true
```

Comment:
One thing I don't like about making this dynamic, is that it'll still get overriden in dev if you've got experimental weights in your settings.

How about using `with-temporary-settig-values` instead?

Reply:
Or mocking the `weights` function directly

Reply:
'with-temporary-settig-values' can't be paralleled in tests, I'm ok with that tho

Reply:
I'll mock the `weights` function with dnamic-redefs

Reply:
I also prefer that route: https://github.com/metabase/metabase/pull/50330/files#diff-bf228cd68e3eafb26461857b7a9e0b1ba8ea2c101eaf051de32008a8448588dbR117-R118